modules/top-level/plugins/mk-plugin-pack.nix

```diff
@@ -40,7 +40,13 @@ let
       postBuild = ''
         find $out -type d -empty -delete
         runHook preFixup
+
+        # Propagate input plugins
+        ${lib.toShellVars { inputs = overriddenPlugins; }}
+        mkdir -p $out/nix-support
+        printf "%s\n" "''${inputs[@]}" >> $out/nix-support/propagated-build-inputs
```

Comment:
It might be possible to use the `recordPropagatedDependencies` command.

https://github.com/NixOS/nixpkgs/blob/b9d43b3fe5152d1dc5783a2ba865b2a03388b741/pkgs/stdenv/generic/setup.sh#L1620C5-L1620C33